### ISSUE TITLE
Ways to log learning rate

### ISSUE THREAD

**Issue by eunsol** [allenai/allennlp#857]
Hi! Is there a simple way to log learning rate as the training progresses? This might be a common use case, so am writing an issue.

**Comment by matt-gardner** [comment]
@DeNeutoy, @matt-peters, is this something that's logged in our tensorboard summaries?

**Comment by matt-peters** [comment]
No it's not currently logged but would be easy to add.

**Comment by DeNeutoy** [comment]
Turns out that this is quite hard to add, because pytorch doesn't keep a running variable of the current learning rate - it just computes it on the fly each time for most optimisers based on the starting learning rate, the step size and whatever learning rate decay you specify.

**Comment by matt-gardner** [comment]
Is this something we can submit a PR to pytorch for?  Is it worth it?

**Comment by schmmd** [comment]
@DeNeutoy is going to open an issue in PyTorch.

**Comment by DeNeutoy** [comment]
Having looked at this I don't think Pytorch are going to be open to adding this sort of state into their optimisers, because some of them don't have a well defined definition of a single learning rate step - e.g LGBFS, or SAGD, so I think unfortunately you'll have to do this manually yourself. If anyone has any better ideas, feel free to re-open!
